staging/src/k8s.io/client-go/rest/config.go

```diff
@@ -504,8 +508,7 @@ func InClusterConfig() (*Config, error) {
 	}
 
 	return &Config{
-		// TODO: switch to using cluster DNS.
-		Host:            "https://" + net.JoinHostPort(host, port),
```

Comment:
Hi,

Please correct me if I am wrong, but, replacing **host** entry here with a static value from a const would need to kill few lines of code and flexibility of using env vars:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/client-go/rest/config.go#L488-L491

However, I understood you tried to fix a **TODO** issue, which I am not even sure the real benefit of replacing a IP addr with a local cluster DNS entry... but, yes, we have a TODO line there.

$container> echo $KUBERNETES_SERVICE_HOST
10.233.0.1

I would just close this PR but feel free to share your point of view.

Reply:
When I read the source code, I found the TODO here and tried to fix it. It seems that it does not improve the current state.

Reply:
@sxllwx thanks!